src/DataStructures/DataBox/DataBox.hpp

```diff
@@ -813,20 +813,21 @@ SPECTRE_ALWAYS_INLINE constexpr void
 db::DataBox<tmpl::list<Tags...>>::mutate_subitem_tags_in_box(
     tmpl::list<Subtags...> /*meta*/) noexcept {
   const auto helper = make_overloader(
-    [this](auto tag_v, std::true_type /*is_compute_tag*/) noexcept {
+    [this](auto tag_v, std::true_type /*is_immutable_item_tag*/) noexcept {
       (void)this;  // Compiler bug warns about unused this capture
       using tag = decltype(tag_v);
       get_deferred<tag>().reset();
     },
-    [this](auto tag_v, std::false_type /*is_compute_tag*/) noexcept {
+    [this](auto tag_v, std::false_type /*is_immutable_item_tag*/) noexcept {
       (void)this;  // Compiler bug warns about unused this capture
       using tag = decltype(tag_v);
       Subitems<ParentTag>::template create_item<tag>(
           make_not_null(&get_deferred<ParentTag>().mutate()),
           make_not_null(&get_deferred<tag>().mutate()));
     });
 
-  EXPAND_PACK_LEFT_TO_RIGHT(helper(Subtags{}, is_compute_tag<ParentTag>{}));
+  EXPAND_PACK_LEFT_TO_RIGHT(
+      helper(Subtags{}, is_immutable_item_tag<ParentTag>{}));
```

Comment:
(optional) I think the overloaded lambda could be quite easily converted to an `if constexpr`.

Reply:
yes, that is coming in a forthcoming PR that changes other things about the function as well